Lib/subprocess.py

```diff
@@ -1460,10 +1483,29 @@ def _posix_spawn(self, args, executable, env, restore_signals):
                         sigset.append(signum)
                 kwargs['setsigdef'] = sigset
 
+            file_actions = []
+            for fd in (p2cwrite, c2pread, errread):
+                if fd != -1:
+                    file_actions.append((os.POSIX_SPAWN_CLOSE, fd))
```

Comment:
Technically, there is no need to explicitly close parent pipe ends because they have `O_CLOEXEC`. One small advantage of doing that is to reduce the number of descriptors before the following `dup2`, which could increase their number (in a corner case when any of descriptors `0, 1, 2` is closed in the parent process) and potentially hit resource limits.

Reply:
Technically, there is no need, but _posixsubprocess does that. In case of doubt, I prefer to mimick _posixsubprocess behavior.

```
    /* Close parent's pipe ends. */
    if (p2cwrite != -1)
        POSIX_CALL(close(p2cwrite));
    if (c2pread != -1)
        POSIX_CALL(close(c2pread));
    if (errread != -1)
        POSIX_CALL(close(errread));
    POSIX_CALL(close(errpipe_read));
```

Reply:
If we decide to skip the close(), I would prefer to also modify _posixsubprocess. But here I only want to use posix_spawn() in more cases, so I prefer to not modify "unrelated" changes :-)